toolkit/scalebar/src/main/java/com/arcgismaps/toolkit/scalebar/internal/ScalebarProperties.kt

```diff
@@ -52,69 +51,89 @@ internal data class ScalebarProperties(
+                scalebarLengthInMapUnits = scalebarLengthInMapUnits,
+                xOffset = displayLength
+            )
+        )
+    } else {
+        val suggestedNumSegments = (displayLength / minSegmentWidth).toInt()
+        val maxNumSegments = minOf(suggestedNumSegments, MAX_NUM_OF_SEGMENTS)
 
-    val numSegments = ScalebarUtils.numSegments(
-        scalebarLengthInMapUnits,
-        maxNumSegments
-    )
+        val numSegments = ScalebarUtils.numSegments(
+            scalebarLengthInMapUnits,
+            maxNumSegments
+        )
 
-    val segmentScreenLength = displayLength / numSegments
-    var currSegmentX = 0.0
-    val localLabels = mutableListOf<ScalebarDivision>()
+        val segmentScreenLength = displayLength / numSegments
+        var currSegmentX = 0.0
+        val localLabels = mutableListOf<ScalebarDivision>()
 
-    localLabels.add(
-        ScalebarDivision(
-            index = -1,
-            xOffset = 0.0 ,
-            labelYOffset = labelTypography.labelStyle.fontSize.value / 2.0,
-            label = "0"
+        // Add the first label at 0
+        localLabels.add(
+            ScalebarDivision(
+                xOffset = 0.0,
+                label = "0"
+            )
         )
-    )
 
-    for (index in 0 until numSegments) {
-        currSegmentX += segmentScreenLength
-        val segmentMapLength: Double = (segmentScreenLength * (index + 1) / displayLength) * scalebarLengthInMapUnits
+        for (index in 1 until numSegments) {
+            currSegmentX += segmentScreenLength
+            val segmentLengthInMapUnits: Double =
+                (segmentScreenLength * index / displayLength) * scalebarLengthInMapUnits
 
-        val segmentText: String = if (index == numSegments - 1) {
-            val displayUnitAbbr = mapUnitsToDisplay.getAbbreviation()
-            "${segmentMapLength.format()} $displayUnitAbbr"
-        } else {
-            segmentMapLength.format()
+            val label = ScalebarDivision(
+                xOffset = currSegmentX,
+                label = segmentLengthInMapUnits.format()
+            )
+            localLabels.add(label)
         }
 
-        val label = ScalebarDivision(
-            index = index,
-            xOffset = currSegmentX,
-            labelYOffset = labelTypography.labelStyle.fontSize.value / 2.0,
-            label = segmentText
+        localLabels.add(
+            generateLastLabel(
+                displayUnitAbbreviation = mapUnitsToDisplay.getAbbreviation(),
+                scalebarLengthInMapUnits = scalebarLengthInMapUnits,
+                xOffset = displayLength
+            )
         )
-        localLabels.add(label)
-    }
 
-    return if (scalebarStyle == ScalebarStyle.Bar || scalebarStyle == ScalebarStyle.Line) {
-        localLabels.takeLast(1)
-    } else {
         localLabels
     }
 }
 
+/**
+ * Generates the last label for the scalebar.
+ *
+ * @since 200.7.0
+ */
+private fun generateLastLabel(
```

Comment:
This function is not generating rather creating a ScalebarDivision and not a label. We can call it Primary Scalebar division. Also it is not the last label in the case of Bar and Line Scalebar.

```suggestion
private fun createPrimaryScalebarDivision(
```

Reply:
good catch 👍🏼